engine/src/main/java/io/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -99,11 +102,26 @@ public DbProcessState(
 
   @Override
   public void putDeployment(final DeploymentRecord deploymentRecord) {
-    for (final ProcessRecord processRecord : deploymentRecord.processes()) {
-      putProcess(processRecord.getKey(), processRecord);
+    for (final DeployedProcessMetadata metadata : deploymentRecord.processes()) {
+      for (final DeploymentResource resource : deploymentRecord.getResources()) {
```

Comment:
This double for isn't ideal, as we want to do the lookup by name, but I guess it should be fine since these collections should be pretty small right? I can't think of a way where we don't have to iterate twice - or could we guarantee something about the ordering? At any rate, it's probably not a big deal since the collections wouldn't be big (or?).

Reply:
Yeah I also thought about it, but I think it is not really guaranteed to be in order and thought this way would be safer. Yes we are anyway limited by the message size.